Explorer/Assets/DCL/Backpack/BackpackCharacterPreviewController.cs

```diff
@@ -0,0 +1,170 @@
+﻿using Arch.Core;
+using Arch.SystemGroups;
+using DCL.AvatarRendering.Wearables.Components;
+using DCL.Backpack.BackpackBus;
+using DCL.Profiles;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.Experimental.Rendering;
+using Avatar = DCL.Profiles.Avatar;
+
+namespace DCL.CharacterPreview
+{
+    public class BackpackCharacterPreviewController : IDisposable
+    {
+        private readonly BackpackCharacterPreviewView view;
+        private readonly ICharacterPreviewFactory previewFactory;
+        private readonly BackpackEventBus backpackEventBus;
+        private readonly CharacterPreviewInputEventBus inputEventBus;
+        private readonly BackpackCharacterPreviewCursorController cursorController;
+
+        private World world;
+        private CharacterPreviewController previewController;
+        private CharacterPreviewModel previewModel;
+
+        public BackpackCharacterPreviewController(BackpackCharacterPreviewView view, ICharacterPreviewFactory previewFactory, BackpackEventBus backpackEventBus, CharacterPreviewInputEventBus inputEventBus)
+        {
+            this.view = view;
+            this.previewFactory = previewFactory;
+            this.backpackEventBus = backpackEventBus;
+            this.inputEventBus = inputEventBus;
+
+            backpackEventBus.EquipEvent += OnEquipped;
+            backpackEventBus.UnEquipEvent += OnUnequipped;
+            backpackEventBus.FilterCategoryByEnumEvent += OnChangeCategory;
+            backpackEventBus.ForceRenderEvent += OnForceRenderChange;
+
+            view.CharacterPreviewInputDetector.OnScrollEvent += OnScroll;
+            view.CharacterPreviewInputDetector.OnDraggingEvent += OnDrag;
+            view.CharacterPreviewInputDetector.OnPointerUpEvent += OnPointerUp;
+            view.CharacterPreviewInputDetector.OnPointerDownEvent += OnPointerDown;
+
+            cursorController = new BackpackCharacterPreviewCursorController(view.CharacterPreviewCursorView, inputEventBus);
+        }
+
+        public void Initialize(Avatar avatar)
+        {
+            previewModel.BodyShape = avatar.BodyShape;
+            previewModel.HairColor = avatar.HairColor;
+            previewModel.SkinColor = avatar.SkinColor;
+            previewModel.ForceRender = new HashSet<string>(avatar.ForceRender);
+
+            //Temporal solution to fix issue with render format in Mac VS Windows
+            if (Application.platform == RuntimePlatform.OSXEditor || Application.platform == RuntimePlatform.OSXPlayer)
```

Comment:
How about to introduce a dedicated utility method that gives us the texture format instead of a full logic duplicate?